src/NServiceBus.Core/Persistence/InMemory/Outbox/InMemoryOutboxSettingsExtensions.cs

```diff
@@ -0,0 +1,24 @@
+namespace NServiceBus.InMemory.Outbox
+{
+    using System;
+    using NServiceBus.Configuration.AdvanceExtensibility;
+    using NServiceBus.Outbox;
+
+    /// <summary>
+    /// Contains InMemoryOutbox related settings extensions.
+    /// </summary>
+    public static class InMemoryOutboxSettingsExtensions
+    {
+        /// <summary>
+        /// Specifies how long the outbox should keep message data in storage to be able to deduplicate.
+        /// </summary>
+        /// <param name="cfg">The outbox settings</param>
+        /// <param name="time">The new duration to be used </param>
+        public static OutboxSettings TimeToKeepDeduplicationData(this OutboxSettings cfg, TimeSpan time)
```

Comment:
Not that I can thing of, `ExposeSettings` is just a convenience class to do this kind of thing so that extenders can get access to the settings to pass things around from an extension method.
So there is no plan to add any extra things to it, but yes we need to be cautious about it to ensure we do not add unintended bloat to `ExposeSettings`.